packages/test/stochastic-test-utils/src/test/random.spec.ts

```diff
@@ -0,0 +1,271 @@
+                    assert.equal(version, 0b100,
+                        `UUID v4 must be version 4, but got '${version}' for bits [48..51].`);
+
+                    const variant = bytes[8] >>> 6;
+                    assert.equal(variant, 0b10,
+                        `UUID v4 must be variant 2, but got '${variant}' for bits [64..65].`);
+
+                    // eslint-disable-next-line unicorn/no-unsafe-regex
+                    const re = /^[\da-f]{8}(?:-[\da-f]{4}){3}-[\da-f]{12}$/;
+                    assert(re.test(uuid),
+                        `UUID must be in the form 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx', but got '${uuid}'.`);
+                }
+            });
+        });
+
+        describe("string", () => {
+            const base58 = "123456789abcdefghijkmnopqrstuvwxyzABCDEFGHJKLMNPQRSTUVWXYZ";
+
+            it("defaults to base58 alphabet", () => {
+                for (const seeds of testSeeds) {
+                    const random = makeRandom(...seeds);
+                    assert_chi2_uniform(
+                        /* generator: */ () => random.string(1),
+                        /* choices: */ base58.split(""),
+                        /* critical: */ 59.48);
+                }
+            });
+        });
+
+        describe("pick", () => {
+            for (const { choices, critical } of [
+                { choices: [1], critical: 0 },
+                { choices: [1, 2], critical: 4.39 },
+                { choices: [1, 2, 3], critical: 3.38 },
+            ]) {
+                it(`of choices ${JSON.stringify(choices)} must have uniform distribution`, () => {
+                    for (const seeds of testSeeds) {
+                        const random = makeRandom(...seeds);
+                        assert_chi2_uniform(() => random.pick(choices), choices, critical);
+                    }
+                });
+            }
+        });
+
+        describe("shuffle", () => {
+            it("accepts empty array", () => {
+                // Paranoid check that 'shuffle()' accepts and does not modify an empty array.
+                const items = [];
+                makeRandom().shuffle(items);
+                assert.deepEqual(items, []);
+            });
+
+            for (const { items, critical } of [
+                { items: [0], critical: 0 },
+                { items: [0, 1], critical: 10.04 },
+                { items: [0, 1, 2], critical: 7.06 },
+            ]) {
+                for (let i = 0; i < items.length; i++) {
+                    it(`each item of ${JSON.stringify(items)} may appears at each position ${i}`, () => {
+                        for (const seeds of testSeeds) {
+                            const random = makeRandom(...seeds);
+
+                            assert_chi2_uniform(
+                                /* generator: */ () => {
+                                    random.shuffle(items);
+                                    return items[0];
```

Comment:
this looks like it should be `items[i]` by the test setup

Reply:
Good catch, thank you!